openwisp_controller/subnet_division/rule_types/base.py

```diff
@@ -76,9 +78,16 @@ def create_subnets_ips(cls, instance, rule_type, **kwargs):
         generated_subnets = cls.create_subnets(
             instance, division_rule, max_subnet, generated_indexes
         )
-        cls.create_ips(instance, division_rule, generated_subnets, generated_indexes)
+        generated_ips = cls.create_ips(
+            instance, division_rule, generated_subnets, generated_indexes
+        )
         SubnetDivisionIndex.objects.bulk_create(generated_indexes)
 
+        if generated_ips:
+            subnet_ips_provisioned.send(
+                sender=SubnetDivisionRule, instance=instance, ip_obj=generated_ips[0]
+            )
```

Comment:
this method could return the generated subnets and ip addresses. The Vpn rule could override this method, eg:

```python
subnets, ip_addresses = super().create_subnets_ips(...)
if ip_addresses:
    auto_ip
```

Reply:
also, I would turn the signal sending a method, and I would pass the generated subnets and ip addresses to it.
Then I wouldn't call this method here, but I would make sure it's called in each rule